### PR TITLE
Add UsePathBaseEnvironment, gets PathBase from Environment Variable

### PR DESCRIPTION
**Pull request by eByte23**
Add UsePathBaseEnvironment, gets PathBase from Environment Variable ASPNETCORE_APPL_PATH

### CODE REVIEW

**Comment by davidfowl** [comment]
Yea this shouldn't be part of HttpAbstractions. This should be part of hosting. It would use a variable set in the config and use this middleware to set the environment.

**Comment by eByte23** [comment]
@davidfowler so should I change this to be a pr on hosting and change middleware accordingly to do something like create a IStartupFilter, similar to the IISIntergration?

Or not bother?

File: src/Microsoft.AspNetCore.Http.Abstractions/Extensions/UsePathBaseExtensions.cs [hunk 1]
```diff
@@ -12,13 +12,14 @@ namespace Microsoft.AspNetCore.Builder
...
     {
+        private static readonly string ServerPath = "APPL_PATH";
         /// <summary>
         /// Adds a middleware that extracts the specified path base from request path and postpend it to the request path base.
         /// </summary>
         /// <param name="app">The <see cref="IApplicationBuilder"/> instance.</param>
         /// <param name="pathBase">The path base to extract.</param>
         /// <returns>The <see cref="IApplicationBuilder"/> instance.</returns>
-        public static IApplicationBuilder UsePathBase(this IApplicationBuilder app, PathString pathBase)
+        public static IApplicationBuilder UsePathBase(this IApplicationBuilder app, PathString pathBase, bool isOptional = false)
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.Http.Abstractions/Extensions/UsePathBaseExtensions.cs]: side note - this is a breaking change.

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http.Abstractions/Extensions/UsePathBaseExtensions.cs]: This doesn't belong here anyways.

**Comment by davidfowl** [comment]
Move this issue to hosting as it has come up more than once. Let's come up with something before sending the PR, I don't think this middleware needs any changes

**Comment by Tratcher** [comment]
https://github.com/aspnet/HttpAbstractions/issues/702#issuecomment-245015384

**Comment by eByte23** [comment]
Create new issue in hosting https://github.com/aspnet/Hosting/issues/849

**Pull request closed without merging**